etcdserver/v3_server.go

```diff
@@ -143,11 +147,17 @@ func (s *EtcdServer) Txn(ctx context.Context, r *pb.TxnRequest) (*pb.TxnResponse
 			warnOfExpensiveReadOnlyTxnRequest(s.getLogger(), start, r, resp, err)
 		}(time.Now())
 
-		get := func() { resp, err = s.applyV3Base.Txn(r) }
-		if serr := s.doSerialize(ctx, chk, get); serr != nil {
-			return nil, serr
+		get := func() error {
+			resp, err = s.applyV3Base.Txn(r)
+			if err != nil {
+				return err
+			}
+			return nil
+		}
+		if err = s.doSerialize(ctx, chk, get); err != nil {
+			return nil, err
 		}
-		return resp, err
+		return resp, nil
```

Comment:
Why change this?

Reply:
> Why change this?

The error here is just going to be nil, It is not necessary to keep err

Reply:
It's a good idea not to hard code `nil` and just use `err` as it originally was here unless there is a reason.

Reply:
ok.